src/bytes/bytes_lemmas.gobra

```diff
@@ -0,0 +1,359 @@
+package bytes
+
+// +gobra
+
+import (
+	. "verification/utils/definitions"
+	// b "verification/utils/bitwise"
+	sl "gobra-libs/byteslice"
+	bytes "gobra-libs/bytes"
+	// seqs "verification/utils/seqs"
+	// sets "verification/utils/sets"
+	. "bytes/spec"
+)
+
+ghost
+requires 0 < p
+requires acc(sl.Bytes(a, 0, len(a)), p)
+requires acc(sl.Bytes(b, 0, len(b)), p)
+requires r == ((unfolding acc(sl.Bytes(a, 0, len(a)), p) in string(a)) == (unfolding acc(sl.Bytes(b, 0, len(b)), p) in string(b)))
+ensures  acc(sl.Bytes(a, 0, len(a)), p)
+ensures  acc(sl.Bytes(b, 0, len(b)), p)
+ensures  r == (sl.View(a) == sl.View(b))
+decreases
+func stringEqualsImplViewEquals(r bool, a, b []byte, p perm)
+
+
+
+// Count
+
+ghost
+requires InRangeInc(idx, 0, len(s))
+requires forall j int :: { j in indices }{ s[j:j+len(sep)] } !(j in indices) ==> !(InRangeInc(j, 0, idx - len(sep))) || s[j:j+len(sep)] != sep || SetContainsInRange(indices, j-len(sep), j)
+requires forall j int :: {s[idx:][j:j+len(sep)]} InRangeInc(j, 0, len(s[idx:]) - len(sep)) ==> s[idx:][j:j+len(sep)] != sep
+ensures  forall j int :: { j in indices }{ s[j:j+len(sep)] } !(j in indices) ==> ( !InRangeInc(j, 0, len(s) - len(sep)) || s[j:j+len(sep)] != sep || SetContainsInRange(indices, j-len(sep), j))
+decreases
+func lemmaCountAux(s, sep seq[byte], indices set[int], idx int) {
```

Comment:
This lemma is very hard to read, and it is hard to get a grasp of what is going on. The fact that we have non trivial assumptions in the body makes it dangerous, from a maintainability standpoint, to have such a lemma. Is there a way to make the pre and postconditions more readable (e.g., by abstracting long and repeated expressions into pure functions whose meaning it is easy to understand)?